api/internal/entity/protocol.go

```diff
@@ -0,0 +1,225 @@
+package entity
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/project-nova/backend/pkg/model"
+)
+
+type Franchise struct {
+	Address            string                 `json:"address"`
+	Name               string                 `json:"name"`
+	VaultAddress       string                 `json:"vaultAddress"`
+	CharacterRegistry  string                 `json:"characterRegistry"`
+	CharacterContracts []*CharacterCollection `json:"characterContract"`
+	StoryRegistry      string                 `json:"storyRegistry"`
+	StoryContracts     []*StoryCollection     `json:"storyContract"`
+	LicenseRepository  string                 `json:"licenseRepository"`
+	LicenseRegistry    string                 `json:"licenseRegistry"`
+}
+
+type CharacterCollection struct {
+	Name    string `json:"name"`
+	Address string `json:"address"`
+}
+
+type StoryCollection struct {
+	Address string `json:"address"`
+	IsCanon bool   `json:"isCanon"`
+}
+
+var FirstFranchise = &Franchise{
```

Comment:
please don't hardcode the contract address in the code. put all the addresses into a config file please.